### DISCUSSION TITLE
de/modes/track/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#16320] [Docs]
# de/modes/track/

Entdecke die effiziente, flexible und anpassbare Multi-Objektverfolgung mit Ultralytics YOLO . Lerne, wie du Videoströme in Echtzeit mühelos verfolgen kannst.

https://docs.ultralytics.com/de/modes/track/

**Comment by MatthiasPSW** [comment]
Does the resume option (resume=True) work in combination with early stopping? Assuming I set the patience to 50, if my training had crashed after 40 epochs without any improvement and I resume the training, will the training stop after another 10 epochs of no improvement, or is the patience timer reset, requiring another full 50 epochs without improvement?

**Reply by glenn-jocher** [reply]
@MatthiasPSW the `resume=True` option does not reset the early stopping patience. If your training resumes after 40 epochs without improvement and you set patience to 50, it will stop after another 10 epochs of no improvement.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics YOLO 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can explore many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. It seems you’re interested in multi-object tracking. You can find relevant information [here](https://docs.ultralytics.com/de/modes/track/).

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If you have a ❓Question about custom training or tracking, please provide detailed information, including dataset examples and logs, and ensure you're following our [Training Tips](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time chat, visit our [Discord](https://ultralytics.com/discord) 🎧. For in-depth discussions, check out [Discourse](https://community.ultralytics.com). Or engage in threads on our [Subreddit](https://reddit.com/r/ultralytics) to share experiences and knowledge.

## Upgrade

Ensure you have the latest version of `ultralytics` and all necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in any of the following environments (with dependencies such as [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) daily.

This is an automated response, and an Ultralytics engineer will assist you soon. 😊
